src/persistence/Elsa.Persistence.EntityFrameworkCore/Extensions/CustomSchemaDbContextBuilderExtensions.cs

```diff
@@ -0,0 +1,178 @@
+using Microsoft.Extensions.DependencyInjection;
+using Npgsql.EntityFrameworkCore.PostgreSQL.Infrastructure;
+
+namespace Elsa.Persistence.EntityFrameworkCore.Extensions
+{
+    public static class CustomSchemaDbContextBuilderExtensions
+    {
+        public static EntityFrameworkCoreElsaBuilder AddCustomSchema(this ElsaBuilder configuration, string schema, string migrationHistoryTableName = null)
+        {
+            EntityFrameworkCoreElsaBuilder frameworkCoreElsaBuilder = new EntityFrameworkCoreElsaBuilder(configuration.Services);
+            if (string.IsNullOrWhiteSpace(schema))
+            {
+                return frameworkCoreElsaBuilder;
+            }
+            DbContextCustomSchema dbContextCustomSchema = new DbContextCustomSchema(schema,
+                    !string.IsNullOrWhiteSpace(migrationHistoryTableName) ? migrationHistoryTableName : DbContextCustomSchema.DefaultMigrationsHistoryTableName);
+
+            configuration.Services.AddSingleton<IDbContextCustomSchema>(dbContextCustomSchema);
+
+            return frameworkCoreElsaBuilder;
+        }
+
+        public static SqliteDbContextOptionsBuilder AddCustomSchemaModelSupport(this SqliteDbContextOptionsBuilder sqliteDbContextOptionsBuilder, DbContextOptionsBuilder dbContextOptionsBuilder, IServiceCollection services)
+        {
+            services.AddTransient<CustomSchemaOptionsExtension>();
+            dbContextOptionsBuilder.ReplaceService<IModelCacheKeyFactory, CustomSchemaModelCacheKeyFactory>();
+            sqliteDbContextOptionsBuilder.AddCustomSchemaExtension(services);
+
+            return sqliteDbContextOptionsBuilder;
+        }
+
+        public static SqlServerDbContextOptionsBuilder AddCustomSchemaModelSupport(this SqlServerDbContextOptionsBuilder sqlServerDbContextOptionsBuilder, DbContextOptionsBuilder dbContextOptionsBuilder, IServiceCollection services)
+        {
+            services.AddTransient<CustomSchemaOptionsExtension>();
+            dbContextOptionsBuilder.ReplaceService<IModelCacheKeyFactory, CustomSchemaModelCacheKeyFactory>();
+            sqlServerDbContextOptionsBuilder.AddCustomSchemaExtension(services);
+
+            return sqlServerDbContextOptionsBuilder;
+        }
+
+        public static NpgsqlDbContextOptionsBuilder AddCustomSchemaModelSupport(this NpgsqlDbContextOptionsBuilder npgsqlDbContextOptionsBuilder, DbContextOptionsBuilder dbContextOptionsBuilder, IServiceCollection services)
+        {
+            services.AddTransient<CustomSchemaOptionsExtension>();
+            dbContextOptionsBuilder.ReplaceService<IModelCacheKeyFactory, CustomSchemaModelCacheKeyFactory>();
+            npgsqlDbContextOptionsBuilder.AddCustomSchemaExtension(services);
+
+            return npgsqlDbContextOptionsBuilder;
+        }
+
+        public static SqliteDbContextOptionsBuilder AddCustomSchemaExtension(this SqliteDbContextOptionsBuilder sqliteDbContextOptionsBuilder, IServiceCollection services)
```

Comment:
These `AddCustomSchemaExtensions` all have the same implementation. Please extract a shared method from it and invoke that one instead?

Reply:
I've extracted a common implementation, but left the three provider specific parameters for ease of use.  They each invoke the common method targeting the IRelationalDbContextOptionsBuilderInfrastructure interface.